pkg/db/canary.go

```diff
@@ -117,20 +135,21 @@ func GetTransformedCheckIDs(canaryID string) ([]string, error) {
 	return ids, err
 }
 
-func UpdateChecksStatus(ids []string, status models.CheckHealthStatus) error {
+func RemoveOldTransformedChecks(ids []string) error {
 	if len(ids) == 0 {
 		return nil
 	}
-	if !utils.Contains(models.CheckHealthStatuses, status) {
-		return fmt.Errorf("invalid check health status: %s", status)
-	}
-	return Gorm.Table("checks").
-		Where("id in (?)", ids).
-		Updates(map[string]any{
-			"status":     status,
-			"deleted_at": gorm.Expr("NOW()"),
-		}).
-		Error
+
+	// Alertmanager checks are marked as healthy on deletion
+	query := `
+        UPDATE checks
+        SET deleted_at = NOW(),
+            status = (CASE WHEN checks.type = 'alertmanager' THEN 'healthy'
+                           ELSE checks.status
+                      END)
+        WHERE id IN (?)
```

Comment:
@moshloop Moving the transformed checks' status setting logic at the query level